flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -686,6 +689,7 @@ void scheduleOrUpdateConsumers(ResultPartitionID partitionId) {
 
 		if (partition.getIntermediateResult().getResultType().isPipelined()) {
 			// Schedule or update receivers of this partition
+			partition.markSomePipelinedDataProduced();
```

Comment:
could this be generalized by having a `partition#markDataProduced` and then calling it fall intermediate results independent of the type?

Reply:
Sure.